renku/service/views/__init__.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku service views."""
+import os
+import re
+from functools import wraps
+
+import jwt
+from flask import request, jsonify
+from git import Repo
+
+from renku.service.config import ALLOWED_EXTENSIONS, CACHE_PROJECTS_PATH, \
+    INVALID_PARAMS_ERROR_CODE, CACHE_UPLOADS_PATH, JWT_KEY, JWT_ALGORITHM
+
+is_git_url = re.compile(
+    r'((git|ssh|http(s)?)|(git@[\w\.]+))(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)?',
+    flags=re.I
+)
+
+
+def allowed_file(filename):
+    """Check if filename is allowed."""
+    has_extension = '.' in filename
+    is_allowed = filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+    return has_extension and is_allowed
+
+
+def find_file(user, file_id):
+    """Search for cached file."""
+    for filename in os.listdir(CACHE_UPLOADS_PATH):
```

Comment:
Thanks, but I need strings not `PosixPaths` for iterations. There is no point in using `iterdir` and then cast every object we get into a string, it's unnecessary. So if there's no other reason for your proposal than readability, I respectfully disagree with this - runtime should never be sacrificed just because somebody finds one way of writing nicer than the other way.

Reply:
Thanks, but no. See below.

Reply:
Sorry if it wasn't clear - my point was not really about readability but that `os.listdir` returns a list while `Path().iterdir()` returns a generator.